### PR TITLE
Hubspot provider addition

### PR DESCRIPTION
**Pull request by vladdoster**
# Submitting Pull Requests
Added HubSpot CRM Oauth2 flow

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must be 100% pep8 and isort clean.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] Feel free to add yourself to `AUTHORS`.
 
 ## Provider Specifics
 
 In case you add a new provider:
 
- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: allauth/socialaccount/providers/hubspot/views.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class HubSpotOAuth2Adapter(OAuth2Adapter):
+    provider_id = HubSpotProvider.id
+    access_token_url = 'https://api.hubapi.com/oauth/v1/token'
+    authorize_url = 'https://app.hubspot.com/oauth/authorize'
+    token_metadata = 'https://api.hubapi.com/oauth/v1/access-tokens/'
+    redirect_uri_protocol = 'https'
+
+    def complete_login(self, request, app, token, **kwargs):
+        resp = requests.get(self.token_metadata + token.token)
+        extra_data = resp.json()
```
Comment (pennersr) [inline allauth/socialaccount/providers/hubspot/views.py]:
```suggestion
        resp.raise_for_status()
        extra_data = resp.json()
```

File: allauth/socialaccount/providers/hubspot/provider.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    pass
+
+
+class HubSpotProvider(OAuth2Provider):
+    id = 'hubspot'
+    name = 'HubSpot'
+    account_class = HubSpotAccount
+
+    def extract_common_fields(self, data):
+        return dict(username=data.get('user'))
```
Comment (pennersr) [inline allauth/socialaccount/providers/hubspot/provider.py]: The mocked response seems to contain a username and id, yet here you are using just `user` ? Is the mocked response not complete, or is this code out of sync?

**Comment by andreasofthings** [comment]
+1

File: allauth/socialaccount/providers/hubspot/provider.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    def extract_uid(self, data):
+        return str(data['id'])
```
Comment (pennersr) [inline allauth/socialaccount/providers/hubspot/provider.py:22]: I checked the API documentation:

https://legacydocs.hubspot.com/docs/methods/oauth2/get-access-token-information

```
{
  "token": "CJSP5qf1KhICAQEYs-gDIIGOBii1hQIyGQAf3xBKmlwHjX7OIpuIFEavB2-qYAGQsF4",
  "user": "test@hubspot.com",
  "hub_domain": "demo.hubapi.com",
  "scopes": [
    "contacts",
    "automation",
    "oauth"
  ],
  "hub_id": 62515,
  "app_id": 456,
  "expires_in": 21588,
  "user_id": 123,
  "token_type": "access"
}
```

There is no mention of an `id` key there.

**Pull request closed without merging**